taskcluster/self_taskgraph/transforms/test.py

```diff
@@ -5,36 +5,30 @@
 
 
 @transforms.add
-def split_by_python_image(config, tasks):
-    python_image_tasks = [
-        task
-        for task in config.kind_dependencies_tasks.values()
-        if task.kind == "docker-image"
-        if task.attributes["image_name"].startswith("py")
-    ]
+def split_by_python_version(config, tasks):
     artifact_dir = "/builds/worker/artifacts"
     for task in tasks:
-        for image_task in python_image_tasks:
-            image_name = image_task.attributes["image_name"]
-
+        for python_version in task.pop("python-versions"):
+            py_name = f"py{python_version.replace('.', '')}"
             new_task = deepcopy(task)
-            new_task["name"] = f"{task['name']}-{image_name}"
+            new_task["name"] = f"{task['name']}-{py_name}"
 
             attributes = new_task.setdefault("attributes", {})
-            attributes["python_version"] = image_name
+            attributes["python_version"] = py_name
 
             worker = new_task.setdefault("worker", {})
-            worker.setdefault("env", {})["MOZ_ARTIFACT_DIR"] = artifact_dir
-            worker["docker-image"] = {"in-tree": image_name}
             worker.setdefault("artifacts", []).append(
                 {
                     "type": "file",
                     "path": f"{artifact_dir}/coverage",
-                    "name": f"public/coverage.{image_name}",
+                    "name": f"public/coverage.{py_name}",
                 }
             )
+            env = worker.setdefault("env", {})
+            env["MOZ_ARTIFACT_DIR"] = artifact_dir
+            env["UV_PYTHON"] = python_version
```

Comment:
This is how the test tasks find the right Python to use.

Reply:
huh neat